### ISSUE TITLE
Default input text theme fails a11y scanner checks

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#58941]
b/157870633

It seems like for label text we are using `grey[600]` which fails the minimum contrast test with the white background. It seems that it needs to be at least grey[700]. I could not find Material specification for this.

Setting as a critical issue as there's no workaround for this. InputDecoration does indeed has `labelStyle` property such as `labelStyle: Theme.of(context).textTheme.bodyText1.copyWith(color: Colors.grey[700]),` .
However, this style will override on every state(unfocused, focused, and error). This means we require a logic to get the state of the TextField. This comes to our second problem. TextField doesn't have a straightforward way to get its own state. This has been a long going discussion here https://github.com/flutter/flutter/issues/48876

**Comment by HansMuller** [comment]
> Setting as a critical issue as there's no workaround for this.

Should this issue have a priority?

**Comment by chunhtai** [comment]
Hi @mehmetf, If we provide the API to set the text color based on different state, will that resolve this issue?

**Comment by Hixie** [comment]
We should just fix the default.

**Comment by chunhtai** [comment]
We currently uses the hint color `const Color(0x8A000000)` for the labelText.

Is there a spec on which text color to use?

**Comment by HansMuller** [comment]
If "just fixing the default" is enough, we should just do that. I'm sure that @willlarche et al. can help with getting the correct default color. 

The issue's description implies the need for specifying different labelStyle colors, depending on the text field's state.

> Setting as a critical issue as there's no workaround for this. InputDecoration does indeed has labelStyle property such as labelStyle: Theme.of(context).textTheme.bodyText1.copyWith(color: Colors.grey[700]), .
> However, this style will override on every state(unfocused, focused, and error). This means we require a logic to get the state of the TextField. This comes to our second problem. TextField doesn't have a straightforward way to get its own state. This has been a long going discussion here #48876

That said, the issue being referred to  (#48876), is about input validation, which need not be roped into this one.

@mehmetf - what's needed to close this issue? I assume that tweaking the floating label's color would be enough?

**Comment by mehmetf** [comment]
> I assume that tweaking the floating label's color would be enough?

Yes. Fixing the default will fix this problem.

**Comment by willlarche** [comment]
@johnsonmh Can you assist?

**Comment by chunhtai** [comment]
A temporary workaround can be overriding the hintcolor for the TextField
```dart
import 'package:flutter/material.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return  MaterialApp(home: new MyHomePage());
  }
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ThemeData data = Theme.of(context).copyWith(hintColor: Colors.green);
    return new Scaffold(
      appBar: AppBar(title: Text('scaffold')),
      body: Column(
        children: [
          // a simple baseline for comparison. scales OK + wraps OK
          Theme(
            data: data,
            child: TextField(
              decoration: InputDecoration(labelText: 'label text'),
            ),
          )
        ],
      ),
    );
  }
}

```

**Comment by chunhtai** [comment]
Talked with @willlarche and @johnsonmh offline. We would like to update the hintcolor to be 60% black(#99000000) in light mode and 60% of white(#99FFFFFF) in dark mode

**Comment by chunhtai** [comment]
After a little bit more investigation, it turns out it may because of the textfield border that failed the accessibility test
![Screen Shot 2020-06-24 at 2 20 20 PM](https://user-images.githubusercontent.com/47866232/85629025-f7766c80-b625-11ea-95b5-c71e19415135.png)

Our default color for border is themeData.colorScheme.onSurface.withOpacity(0.38).  We do have the ability to override the color
```
TextField(decoration: InputDecoration(labelText: 'test', enabledBorder: UnderlineInputBorder(borderSide: BorderSide(width: 1, color: Colors.black))),)
```
and i believe that should solve the customers use case. @HansMuller do you think we should fix the default color for the border in this case?

**Comment by willlarche** [comment]
I would prefer to do these minor fixes if possible since they can make a
slight impact in a11y support.

**Comment by chunhtai** [comment]
Current status: Waiting for the customer to confirm the textfield border is actually the problem in their test.

**Comment by chunhtai** [comment]
Closed as there will be no action item for flutter team, the issue is is on the accessibility scanning tool. internal tracking bug
b/79165151
